x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/JreHttpUrlConnection.java

```diff
@@ -176,6 +176,16 @@ private boolean shouldParseBody(int responseCode) {
         }
         SqlExceptionType type = SqlExceptionType.fromRemoteFailureType(failure.type());
         if (type == null) {
+            if (con.getResponseCode() == HttpURLConnection.HTTP_BAD_REQUEST) {
+                return new ResponseOrException<>(new SQLException("It doesn't look like the X-Pack or the X-Pack SQL component"
```

Comment:
Slight rephrasing, removing some `the` and typos: `X-Pack/ SQL do not seem to be available on the Elasticsearch node using the access path '...' . Please verify X-Pack is installed and SQL enabled. Alternatively, check if any proxy is interfering the communication to Elasticsearch.`

Reply:
Thanks. I modified it.